grakn-engine/src/main/java/ai/grakn/engine/backgroundtasks/distributed/DistributedTaskManager.java

```diff
@@ -95,9 +95,7 @@ public String scheduleTask(BackgroundTask task, String createdBy, Instant runAt,
                 .interval(period)
                 .configuration(configuration);
 
-        stateStorage.newState(taskState);
-
-        producer.send(new ProducerRecord<>(NEW_TASKS_TOPIC, taskState.getId(), configuration.toString()));
+        producer.send(new ProducerRecord<>(NEW_TASKS_TOPIC, taskState.getId(), TaskState.serialize(taskState)));
```

Comment:
If this method is about inserting new tasks, it should be renamed. In our context "schedule task" has a specific so this name leads to confusion.